src/NUnitFramework/framework/Assert.cs

```diff
@@ -360,19 +360,18 @@ private static void ReportFailure(ConstraintResult result, string message, param
 
         private static void ReportFailure(string message)
         {
-            // Failure is recorded in <assertion> element in all cases
-            TestExecutionContext.CurrentContext.CurrentResult.RecordAssertion(
-                AssertionStatus.Failed, message, GetStackTrace());
-
             // If we are outside any multiple assert block, then throw
             if (TestExecutionContext.CurrentContext.MultipleAssertLevel == 0)
                 throw new AssertionException(message);
+
+            // Otherwise, just record the failure in an <assertion> element
+            var result = TestExecutionContext.CurrentContext.CurrentResult;
```

Comment:
I've also been looking at this issue. If I can't come up with something better, I'll add your suggested fix here before merging.